testlapack/dlartg.go

```diff
@@ -8,29 +8,81 @@ import (
 	"math"
 	"math/rand"
 	"testing"
+
+	"github.com/gonum/floats"
 )
 
 type Dlartger interface {
 	Dlartg(f, g float64) (cs, sn, r float64)
 }
 
 func DlartgTest(t *testing.T, impl Dlartger) {
+	const (
+		tol   = 1e-14
+		safmn = 2.002083095183101e-146
+		safmx = 4.994797680505588e+145
+	)
 	rnd := rand.New(rand.NewSource(1))
-	for i := 0; i < 100; i++ {
-		f := rnd.NormFloat64()
-		g := rnd.NormFloat64()
+	for i := 0; i < 1000; i++ {
+		var (
+			f, g         float64
+			fHuge, gHuge bool
+		)
+		p := rnd.Float64()
+		switch {
+		case p < 1/3.0:
+			// Huge f.
+			fHuge = true
+			f = math.Pow(10, 10-20*rnd.Float64()) * safmx
+			if rnd.Float64() < 0.5 {
+				f *= -1
+			}
+		case p < 2/3.0:
```

Comment:
It would be probably slightly clearer if I wrote what you suggest. I will change it.